### PR TITLE
feat(core): add sign-in timestamp in session interaction

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Add sign-in timestamp to session interaction. Originally introduced in #1917 but got reverted for we didn't think it clearly enough back in the day.

## Testing

- [x] UT

### CODE REVIEW

File: packages/core/src/lib/session.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import dayjs from 'dayjs';
```
Comment (gao-sun) [inline packages/core/src/lib/session.ts:1]: prefer `date-fns` over `dayjs`. see #2343

File: packages/core/src/lib/session.ts [hunk 2]
```diff
@@ -14,20 +15,30 @@ export const assignInteractionResults = async (
...
   const redirectTo = await provider.interactionResult(
     ctx.req,
     ctx.res,
     {
-      // Merge with current result
-      ...details?.result,
-      ...result,
+      ...mergedResult,
+      login: mergedResult.login
+        ? {
```
Comment (gao-sun) [inline packages/core/src/lib/session.ts:30]: `conditional()`

File: packages/core/src/lib/session.ts [hunk 3]
```diff
@@ -14,20 +15,30 @@ export const assignInteractionResults = async (
...
+            ...mergedResult.login,
+            // Update ts(timestamp) if the accountId has been set in result
+            ts: result.login?.accountId ? ts : mergedResult.login.ts,
```
Comment (gao-sun) [inline packages/core/src/lib/session.ts:33]: is there any specific reason we use `ts` over `timestamp`?

Comment (charIeszhao) [reply packages/core/src/lib/session.ts:33]: image

**Review by darcyYe — APPROVED** [review]

File: packages/core/src/lib/session.ts (hunk 1 above)
Comment (charIeszhao) [reply packages/core/src/lib/session.ts]: done

**Pull request merged**